Yoast/javascript
# Refactor integration

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Implements the necessary steps to integrate the new yoastseo structure with the plugin.

## Relevant technical choices:
* We don't set a new researcher on `app.researcher` anymore, because this wasn't called anywhere. 
* No new researcher is created anymore when a new paper is set, since it's possible to switch papers on the existing researcher.
* Running the new structure in the content analysis app: We needed to ensure that the locale is properly set when the worker loads, which is before the locale input field loads. Therefore the input saves the value to `localStorage`, then `refreshes` the page. On initializing the web worker, the value is fetched from localStorage and used to set the appropriate researcher (defaulting to the default researcher if it doesn’t exist).
* For reasons of consistency, we always pass the researcher in all assessors.
* Some imports and exports are adjusted to reflect the new file structure.

## Test instructions

This PR can be tested by following these steps:

* Start the content analysis app.
* Set the locale to `en_US`.
* Confirm that you get assessment results. (Note that some assessments, like passive voice and keyphrase length currently error. This should be fixed in a following PR.)
* Make sure that the assessments use language-specific configuration, e.g.:
  * Add a sentence with a transition word such as `however`.
  * The transition words assessments should recognize the transition word and mark it.
  * Change the locale to another language such as `fr_FR`.
  * Add a sentence with a transition word from that language, e.g. `alors`.
  * The transition words assessments should recognize the transition word and mark it.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-651
